h/presenters/annotation_searchindex.py

```diff
@@ -16,6 +16,7 @@ def asdict(self):
         docpresenter = DocumentSearchIndexPresenter(self.annotation.document)
 
         result = {
+            'authority': self.annotation.userid,
```

Comment:
If we split this apart in the Python code (for which we have the [`split_user` function](https://github.com/hypothesis/h/blob/156d9aa10c3f12ef94afb21d6354817298dbbce1/h/util/user.py#L6-L21)), we could remove the need for extra complexity in the Elasticsearch configuration. Would that work?

Reply:
For reasons outlined in [this comment](https://github.com/hypothesis/h/pull/4700#issuecomment-347519407) I do think we need to do the filtering in ES rather than mixing it between ES and Python.